opennms-correlation/drools-correlation-engine/src/main/java/org/opennms/netmgt/correlation/drools/DroolsCorrelationEngine.java

```diff
@@ -270,7 +273,13 @@ public void initialize() throws Exception {
         }
 
         if (factObjects != null) {
-            factObjects.forEach(fact -> m_kieSession.insert(fact));
+            factObjects.forEach(fact -> {
+                try {
+                    m_kieSession.insert(streamIn(fact, m_kieSession.getClass().getClassLoader()));
```

Comment:
DroolsStreamUtils#streamIn throws a ClassNotFoundException for facts that are declared in and compiled from DRL.
I have not yet found a ClassLoader that can find those classes.